src/jit/codegenxarch.cpp

```diff
@@ -7393,11 +7427,25 @@ CodeGen::genCkfinite(GenTreePtr treeNode)
     // If exponent is all 1's, throw ArithmeticException
     genJumpToThrowHlpBlk(EJ_je, SCK_ARITH_EXCPN);
 
+#ifdef _TARGET_AMD64_
     // if it is a finite value copy it to targetReg
     if (treeNode->gtRegNum != op1->gtRegNum)
     {
         inst_RV_RV(ins_Copy(targetType), treeNode->gtRegNum, op1->gtRegNum, targetType);
     }
+#else // !_TARGET_AMD64_
+    if (targetType == TYP_DOUBLE)
+    {
+        inst_RV_RV_IV(INS_shufps, EA_16BYTE, targetReg, targetReg, 0xb1);
+    }
+    else
+    {
+        if (treeNode->gtRegNum != op1->gtRegNum)
+        {
+            inst_RV_RV(ins_Copy(targetType), targetReg, op1->gtRegNum, targetType);
+        }
+    }
+#endif // !_TARGET_AMD64_
```

Comment:
IMO,  #ifdefs at multiple places is reducing the readability of logic and also one has to be careful while making fixes/improvements to this routine in future.  My suggestion is structure this on the below lines

```
// common to x86 and x64
GenTree* op1 = treeNode->GetOp1();
regNumber op1Reg = genConsumeReg(op1);
regNumber targetReg = treeNode->gtRegNum;
var_types targetType = treeNode->TypeGet();
int expMask = (targetType == TYP_FLOAT) ? 0x7F800000 : 0x7FF00000;

// Extract reserved reg 
assert(treeNode->gtRsvdRegs != RBM_NONE);
assert(genCountBits(treeNode->gtRsvdRegs) == 1);
regNumber tmpReg = genRegNumFromMask(treeNode->gtRsvdRegs);

#ifdef _TARGET_64BIT_
 // x64 comple implementation goes here
# else
 // x86 complete implementation goes here
# endif

genProduceReg(treeNode);

```